File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
let [...arr] = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");
let [n, m, v] = arr.shift().split(" ").map(Number);
arr = arr.map((v) => v.split(" ").map(Number));
let adj = Array.from(Array(n + 1), () => new Array());
let vis = Array(n + 1).fill(0);
let q = [];
let qIdx = 0;

for (let i = 0; i < m; i++) {
  adj[arr[i][0]].push(arr[i][1]);
  adj[arr[i][1]].push(arr[i][0]);
}
for (let i = 1; i < n + 1; i++) adj[i].sort((a, b) => a - b);
let answer = [];

/** DFS */
let d = [];
let dfs = (v) => {
  d.push(v);
  for (let i = 0; i < adj[v].length; i++) {
    if (vis[adj[v][i]] !== 0) continue;
    vis[adj[v][i]] = vis[v] + 1;
    dfs(adj[v][i]);
  }
};
vis[v] = 1;
dfs(v);

/** BFS */
vis.fill(0);
let b = [v];
vis[v] = 1;
for (let i = 0; i < adj[v].length; i++) {
  q.push(adj[v][i]);
  vis[adj[v][i]] = 2;
}

while (q.length > qIdx) {
  let cur = q[qIdx++];
  b.push(cur);
  for (let i = 0; i < adj[cur].length; i++) {
    if (vis[adj[cur][i]] !== 0) continue;
    q.push(adj[cur][i]);
    vis[adj[cur][i]] = vis[cur] + 1;
  }
}

answer.push(d.join(" "));
answer.push(b.join(" "));
console.log(answer.join("\n"));

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <bits/stdc++.h>
using namespace std;
int n, m, v;
vector<vector<int>> adj(1010);
vector<bool> vis(1010);
void bfs() {
  vis.assign(1010, 0);
  queue<int> q;
  q.push(v);
  vis[v] = 1;
  while (!q.empty()) {
    int cur = q.front();
    cout << cur << " ";
    q.pop();
    for (int nxt : adj[cur]) {
      if (vis[nxt]) continue;
      vis[nxt] = 1;
      q.push(nxt);
    }
  }
  cout << "\n";
}
void dfs(int s) {
  cout << s << " ";
  vis[s] = 1;
  for (int nxt : adj[s]) {
    if (vis[nxt]) continue;
    dfs(nxt);
  }
}

int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);

  cin >> n >> m >> v;
  while (m--) {
    int a, b;
    cin >> a >> b;
    adj[a].push_back(b);
    adj[b].push_back(a);
  }
  for (int i = 1; i <= n; i++)
    sort(adj[i].begin(), adj[i].end());

  dfs(v);
  cout << "\n";
  bfs();
}